src/js/core/widget/core/OnOffSwitch.js

```diff
@@ -386,6 +399,136 @@
 
 				if (self._type === "select") {
 					element.selectedIndex = value;
+					self._ui.input.checked = !!value;
+				}
+			};
+
+			OnOffSwitch.prototype._onDragStart = function () {
+				var self = this,
+					ui = self._ui,
+					checkbox = ui.input,
+					handler = ui.handler,
+					inputRect = checkbox.getBoundingClientRect(),
+					handleRect = handler.getBoundingClientRect(),
+					inputStyle = window.getComputedStyle(checkbox),
+					border = parseInt(inputStyle.borderLeftWidth, 10) +
+						parseInt(inputStyle.borderLeftWidth, 10);
+
+				self._moveWidth = inputRect.width - handleRect.width + border;
```

Comment:
If you use border only in this function, you may want to fill it out immediately.

Reply:
Ok. I'll refactor this.

Reply:
Done.